modules/metastore/src/LifeCycle/LifeCycle.php

```diff
@@ -115,9 +117,57 @@ public function __construct(
     $this->configFactory = $configFactory;
   }
 
+  /**
+   * Check if the entity is part of the metastore.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   A Drupal content entity.
+   *
+   * @return bool
+   *   Returns true if the entity is used by the metastore.
+   */
+  public function entityIsValidItem(EntityInterface $entity) {
+    // Currently, all metastore items are Data nodes. We can short-circuit here
+    // based on this. When we start expecting other entity types we can modify
+    // this logic.
+    // @todo Modify this logic when other entity types are valid.
+    if ($entity->getEntityTypeId() == 'node' && $entity->bundle() === 'data') {
+      // @todo Inject this.
+      $storageClass = \Drupal::service('dkan.metastore.storage')::getStorageClass();
+
+      // If the storage class used implements the entity storage interface,
+      // continue.
+      // @todo Should we look at the entity's storage class instead?
+      if (!is_a($storageClass, MetastoreEntityStorageInterface::class, TRUE)) {
+        return FALSE;
+      }
+
+      // @todo Inject these.
+      $storageEntityType = \Drupal::service('dkan.metastore.metastore_item_factory')::getEntityType();
+      $storageBundles = \Drupal::service('dkan.metastore.metastore_item_factory')::getBundles();
+
+      // If the type and bundle are correct, return true.
+      if ($entity->getEntityTypeId() != $storageEntityType) {
+        return FALSE;
+      }
+      if (in_array($entity->bundle(), $storageBundles)) {
+        return TRUE;
+      }
+    }
+    return FALSE;
```

Comment:
```suggestion
    if ($entity->getEntityTypeId() !== 'node' || $entity->bundle() !== 'data') {
      return FALSE;
    }
    return TRUE;
```
If we're just going to hard code "node" and "data" anyway all the other stuff is redundant I think?

OTOH the whole function is kind of redundant because the `wrap()` method will instantiate a new `Data` object, which validates the entity in its constructor with exactly the same criteria. The issue is that that validation function doesn't return true or false, it throws an exception if invalid and does nothing if valid. Using a try/catch around `wrap()` in those entity hook functions is a icky, so an idea I had was:

a) Make `Data::validate()` public and static, and have it return `bool` rather than `void`/throw.
b) throw the exception in the constructor if validate fails.
c) Call Data::validate in the .module file rather than your new `LifeCycle::entityIsValidItem()`. 

It's not exactly a slam-dunk. This whole area is not great. Maybe I'm missing another cleaner approach. But at minimum let's simplify that validation function.